lumigator/frontend/src/sdk/experimentsService.ts

```diff
@@ -101,6 +101,7 @@ export async function createExperimentWithWorkflows(
         experiment_id: experimentId,
         model: model.model,
         provider: model.provider,
+        secret_key_name: `${model.provider}_api_key`,
```

Comment:
what about models that don't require an api key, would this break something in the backend? maybe we can only add it to the payload only if its an api key one

maybe something like this?

```suggestion
...(model.requirements.includes('api-key') && {secret_key_name: `${model.provider}_api_key`}),
```

Reply:
on another note I also think its a bit leaky to expect the sdk clients or frontend to specify which api key to use for each model, can't we just use the same naming convention on the backend so we don't have to pass it through the api? afterall the frontend can't just create new keys, but only one of the 4 ones we currently support.

it sounds like we could have some immutable property or something on each key that tells the backend which model its for, or some relationship between models and the secrets tables

Reply:
Hey, thanks @khaledosman. I like the suggestion to check if the requirements say an API key is needed! 

RE: Being leaky:

Some folks could create secrets which don’t follow our UI (and soon SDK) naming, so we have to be flexible in allowing the caller to say which secret key contains the API key they need to be passed through to the job (even via the workflow).

Reply:
wouldn't passing a model id when creating the secret do the trick? that way we can infer the model via the secret and vice versa.